### DISCUSSION TITLE
ESP32: Difference between LEDLink and LEDLink_i ?

### DISCUSSION THREAD

**Discussion by sigma415** [arendst/Tasmota#22072] [Q&A] [answered]
Hello,

what is the difference between LEDLink and LEDLink_i ?

I connected ESP32 GPIO12 to LED via resistor, because I want to **indicate an established WiFi connection**.
But GPIO12 behaviour is identical regardless if I assign LEDLink or LEDLink_i 
And the funny thing in both cases is: 
After Reset most times LED goes off after blinking, but sometimes it also stays on !? WiFi connection is established in both cases.

I use precompiled image tasmota32.bin v14.2.0

How can I realize that LED is on after successfull WiFi connection ?
Thanks a lot.

**Comment by s-hadinger** [comment]
They are inverted. Obviously when blinking it does not make a big difference. 

It depends whether you prefer a solid light or no light when wifi or mqtt is connected

**Reply by sigma415** [reply]
> They are inverted. Obviously when blinking it does not make a big difference.
> 
> It depends whether you prefer a solid light or no light when wifi or mqtt is connected

Thanks. What you describe is what I had expected, but it doesn't react like that. 
I had described above in detail, what is really happening.

**Reply by sfromis** [reply]
In standard setup, blinking does not merely indicate lack of Wifi connection, but is also when missing MQTT connection, if configured. What you have in `LedState` and `SetOption31` can also influence behavior.
https://tasmota.github.io/docs/Lights/#status-leds

**Comment by sigma415** [comment]
Additionally I had used 4 GPIOs as Led1 ... Led4 and had controlled them with `LedPower(x)` out of a rule. 
But there seems to be some mysterious interference between LedLink and Led(x) when using `LedPower(x)` to controll the Led(x).
I changed now Led1 ... 4 to Relay1 ... 4 and control them with normal `Power(x) `command and set `LedPower 1 `(which also internally sets `LedState 8`)
Now LedLink (not LedLink_i) stays on with WiFi connected. :)

**Reply by sfromis** [reply, marked as answer]
If you want full control yourself, using `Led` gpio config and the `LedPower` command is not the intended way, as status leds (do read the link I posted) are meant to primarily be managed by Tasmota. Configuring as a Relay is indeed how to control a simple led, or you can config it as a PWM if you want to manage brightness too.
